bin/grommet.js

```diff
@@ -17,76 +18,159 @@ String.prototype.capitalize = function() {
+    .pipe(installDependencies());
 }
 
 var tasks = {
-  'init': function(done) {
-    if (!nodeVersionSupported() || !npmVersionSupported(npmVersion)) {
-      console.error('[grommet] Grommet requires Node v0.10+ and NPM 1.4.x+.');
-      console.error(
-        '[grommet] Currently you have Node ' + process.version +
-        ' and NPM ' + npmVersion
-      );
-      process.exit(1);
+  'init': function(command) {
+    return function() {
+      if (!dependenciesSupported(command.env.nodeVersion, command.env.npmVersion)) {
+        process.exit(1);
+      }
+
+      var newAppPath = path.join(process.cwd(), command.options.app);
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.log('Error trying to create project: ' + err);
+        } else {
+          var templateFolder = path.join(command.env.templatePath, 'init/**');
+          var mobileIcon = path.join(command.env.grommetPath, 'img/mobile-app-icon.png');
+          var shortcutIcon = path.join(command.env.grommetPath, 'img/shortcut-icon.png');
+
+          gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
+          gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        }
+      })
     }
+  },
+  'new': function(command) {
+    return function() {
+      if (!dependenciesSupported(command.env.nodeVersion, command.env.npmVersion)) {
+        process.exit(1);
+      }
+
+      var newAppPath = path.join(process.cwd(), command.options.app);
 
-    mkdirp('./' + app, function(err) {
-      if (err) {
-        console.log('Error trying to create project: ' + err);
-      } else {
-        process.chdir('./' + app);
-
-        var templateFolder = path.join(grommetPath, 'templates/init/**');
-        var mobileIcon = path.join(grommetPath, 'img/mobile-app-icon.png');
-        var shortcutIcon = path.join(grommetPath, 'img/shortcut-icon.png');
-
-        gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(templateFolder, { dot: true })
-        .pipe(template({
-          appName: app,
-          appTitle: title,
-          grommetVersion: packageJSON.version
-        }))
-        .pipe(gulp.dest('./'))
-        .pipe(install()).on('finish', function() {
-          done();
-        });
+      if (fileExists(newAppPath)) {
+        console.error('[grommet] Error while creating new app. Directory "'+command.options.app+'" already exists.');
+        process.exit(1);
       }
-    });
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.error('[grommet] Error trying to create project: ' + err);
+          process.exit(1);
+        }
+
+        try {
+          var templateFolder = path.join(command.env.templatePath, 'new/**');
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        } catch(err) {
+          shelljs.rm('-rf', newAppPath);
```

Comment:
i like that we delete the project folder if anything fails. can we add that also inside the init command?